src/Psl/Filesystem/Internal/change_group.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Filesystem\Internal;
+
+use Psl\Filesystem;
+use Psl\Filesystem\Exception;
+use Psl\Internal;
+use Psl\Str;
+
+use function chgrp;
+use function lchgrp;
+
+/**
+ * @param iterable<string>  $files
+ * @param string|int        $group
+ *
+ * @throws Exception\RuntimeException   If unable to change the ownership for
+ *                                      the given file.
+ *
+ * @internal
+ */
+function change_group(iterable $files, $group, bool $recursive = false): void
```

Comment:
What about splitting into:

- change_group
- change_group_recursive

Reply:
I'm having my doubts here, i think a better option would be to get rid of the recursive flag.

i had so many problems with `change_permissions` recursively, that i think it's better not to allow it.

e.g:

assuming we have directory `foo`, which contains `bar.txt`, and `baz.txt`.

if `foo` is readable, and we want to make it not readable recursively, we have to first read the directory, change the permissions of the files, then the change the permissions of `foo` itself.

however, if `foo` is not readable, and we want to make it readable, we have to do the complete opposite, i.e: change permissions of `foo` to readable, then all of it's content recursively.

another issue is that we can't change anything recursively if `foo` is not readable, and the permission we want to set also doesn't allow reading.

Reply:
Wow .. I am not sure it's worth the hassel. The amount of times I ever used `change_groups` in PHP is around 0.